src/index.js

```diff
@@ -288,71 +269,138 @@ class DropdownTreeSelect extends Component {
     }
   }
 
-  render() {
-    const { disabled, readOnly, mode, texts, inlineSearchInput } = this.props
-    const { showDropdown, currentFocus, tags } = this.state
-
-    const activeDescendant = currentFocus ? `${currentFocus}_li` : undefined
-
-    const commonProps = { disabled, readOnly, activeDescendant, texts, mode, clientId: this.clientId }
-
-    const searchInput = (
-      <Input
-        inputRef={el => {
-          this.searchInput = el
-        }}
-        onInputChange={this.onInputChange}
-        onFocus={this.onInputFocus}
-        onBlur={this.onInputBlur}
-        onKeyDown={this.onKeyboardKeyDown}
-        {...commonProps}
-      />
-    )
-    return (
+  useEffect(() => {
+    treeManager.current = new TreeManager({
+      data,
+      mode,
+      showPartiallySelected,
+      rootPrefixId: clientId,
+      searchPredicate,
+    })
+
+    setState(prevState => {
+      const currentFocusNode = prevState.currentFocus && treeManager.current.getNodeById(prevState.currentFocus)
+      if (currentFocusNode) {
+        currentFocusNode._focused = true
+      }
+      return {
+        showDropdown: /initial|always/.test(showDropdownProp) || prevState.showDropdown === true,
+        ...treeManager.current.getTreeAndTags(),
+      }
+    })
+
+    return () => document.removeEventListener('click', handleOutsideClick, false)
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [JSON.stringify(data), mode, showDropdownProp, showPartiallySelected, searchPredicate])
+
+  useEffect(() => {
+    // register event listeners only if there is a state change
+    if (state.showDropdown) {
+      document.addEventListener('click', handleOutsideClick, false)
+    } else {
+      document.removeEventListener('click', handleOutsideClick, false)
+    }
+  }, [handleOutsideClick, state.showDropdown])
+
+  const { showDropdown, currentFocus, tags } = state
+
+  const activeDescendant = currentFocus ? `${currentFocus}_li` : undefined
+
+  const commonProps = { disabled, readOnly, activeDescendant, texts, mode, clientId }
+
+  const searchInput = (
+    <Input
+      ref={searchInputRef}
```

Comment:
I had some problems passing the ref as `inputRef`. It seems like AVA can't deal with refs passed through custom named props.

Reply:
Trying to understand here - don't we do that already? What was the exact error/issue you ran into?

Reply:
The error I got when using `inputRef` is: 

```
 PointerLookupError {
    index: 10,
    message: 'Could not deserialize buffer: pointer 10 could not be resolved',
}
```
I found this relatated [issue](https://github.com/concordancejs/concordance/issues/66). I tried to upgrade AVA to the latest version but it didn't solve the problem.